--- original.sh	2017-07-19 13:12:18.721675400 +1000
+++ patched.sh	2017-07-19 13:11:16.247067800 +1000
@@ -28,7 +28,7 @@
 fi
 
 # allow the container to be started with `--user`
-if [ "$1" = 'postgres' ] && [ "$(id -u)" = '0' ]; then
+if [ "$(id -u)" = '0' ]; then
 	mkdir -p "$PGDATA"
 	chown -R postgres "$PGDATA"
 	chmod 700 "$PGDATA"
@@ -47,7 +47,7 @@
 	exec gosu postgres "$BASH_SOURCE" "$@"
 fi
 
-if [ "$1" = 'postgres' ]; then
+#if [ "$1" = 'postgres' ]; then
 	mkdir -p "$PGDATA"
 	chown -R "$(id -u)" "$PGDATA" 2>/dev/null || :
 	chmod 700 "$PGDATA" 2>/dev/null || :
@@ -140,6 +140,6 @@
 		echo 'PostgreSQL init process complete; ready for start up.'
 		echo
 	fi
-fi
+#fi
 
 exec "$@"